frontend/src/libs/components/card/card.tsx

```diff
@@ -18,33 +21,41 @@ const Card: React.FC<Properties> = ({
   onClick,
   isActive,
   iconName,
+  iconColor = 'currentColor',
+  statusIcon,
 }) => {
+  const shouldShowStatusIcon = isActive && statusIcon;
+  const hasNoImageOrIcon = !imageUrl && !iconName;
+
   return (
     <button
       className={getValidClassNames(
         styles['item'],
         isActive && styles['selected'],
+        hasNoImageOrIcon && styles['no-image'],
       )}
       onClick={onClick}
     >
-      <div
-        className={getValidClassNames(
-          styles['name'],
-          !imageUrl && styles['no-image'],
-        )}
-      >
-        {imageUrl && (
-          <div className={styles['image-placeholder']}>
-            <img src={imageUrl} alt="not found" className={styles['image']} />
-          </div>
-        )}
-        {iconName && (
-          <div className={styles['image-placeholder']}>
-            <Icon name={iconName} />
-          </div>
-        )}
-        {title}
-      </div>
+      {!hasNoImageOrIcon && (
+        <div className={styles['image-container']}>
+          {imageUrl && (
+            <div className={styles['image-placeholder']}>
+              <img src={imageUrl} alt="not found" className={styles['image']} />
+            </div>
+          )}
+          {iconName && (
+            <div className={styles['icon-background']}>
+              <Icon name={iconName} color={iconColor} />
+            </div>
+          )}
```

Comment:
imageUrl  and iconName should be replaced with boolean bcos it would render it instead of jsx component

Reply:
That makes sense, working on it